Jenkinsfile

```diff
@@ -245,6 +246,55 @@ _stage('Code Quality', context) {
 } //end stage
 
 
+_stage('ZAP Security Scan', context) {
+    podTemplate(
+        label: 'owasp-zap${context.uuid}',
+        name: 'owasp-zap${context.uuid}',
+        serviceAccount: 'jenkins',
+        cloud: 'openshift',
+        containers: [
+            containerTemplate(
+                name: 'jnlp',
+                image: '172.50.0.2:5000/openshift/jenkins-slave-zap',
+                resourceRequestCpu: '500m',
+                resourceLimitCpu: '1000m',
+                resourceRequestMemory: '3Gi',
+                resourceLimitMemory: '4Gi',
+                workingDir: '/home/jenkins',
+                command: '',
+                args: '${computer.jnlpmac} ${computer.name}'
+            )
+        ]
+    ) {
+        node('master') {
+            //the checkout is mandatory
+            echo "checking out source"
+            echo "Build: ${BUILD_ID}"
+            checkout scm
+            dir('zap') {
+                def retVal = sh (
+                    script: """
+                        set -eux
+                        ./runzap.sh
+                    """,
+                    returnStatus: true
```

Comment:
Notice that the exit value is not being analyzing and properly throwing exception when an error occurred (`reval != 0`):
see [](https://jenkins.io/doc/pipeline/steps/workflow-durable-task-step/#code-sh-code-shell-script)